### PR TITLE
Adt 111 populate prompts

### PR DESCRIPTION
**Pull request by alistairknight**
# **What this PR does / why we need it**:
*A brief description of changes being made*
*Link to the relevant work items: e.g: Relates to ADO Work Item AB#213700 and builds on #3376 (link to ADO Build ID URL)*

# **Special notes for your reviewer**
*Any specific actions or notes on review?*

# Testing
*Any relevant testing information and pipeline runs.*

# Checklist (please delete before completing or setting auto-complete)
- [ ] Story Work items associated (not Tasks)
- [ ] Successful testing run(s) link provided
- [ ] Title pattern should be `{work item number}: {title}`
- [ ] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests
- [ ] Version number in **package.json** has been updated

# **How does this PR make you feel**:
gif

### CODE REVIEW

**Comment by alistairknight** [comment]
In order to populate the models and prompts tables, use the following postman requests:

1. Creating models:

`POST` to `http://localhost:3006/models`

`{
    "deploymentName": "Bedrock",
    "deploymentVendor": "AWS",
    "deploymentService": "AWS",
    "model": "Bedrock",
    "type": "chat"
}`

Add a few different models and providers to see the full range in the front end, e.g. AWS, OpenAI, Google etc.

2. Creating prompts:
`POST` to `localhost:3006/prompts`

`{
    "project": "mcu",
    "name": "Ministerial correspondence",
    "modelId": "AWS:AWS:Bedrock",
    "prompt": "[INST]\nYou are an expert in writing responses to correspondence from the general public. You are assisting a Defra employee in writing a response to the document contained in [DOCUMENT] below.\n\nYou should only provide a response based on the knowledge contained in [CONTEXT] below. You should not infer any additional information.\n\nYour response should always be in the format of an email or a letter. You should always sign the response with 'Yours sincerely, [NAME] [ROLE]'.\n\nIf you are unable to find any relevant information, you should respond with a message stating that you are unable to provide a response.\n\nYou may be provided with a previous response in [PREVIOUS_RESPONSE] below. Any requests in [REQUESTS] should be considered within the context of the previous response.\n\n[REQUESTS] should only be the operator asking for amendments to your [PREVIOUS_RESPONSE]. E.g. the operator asks you to include more information about a specific topic.\n\nUnless overriden in [REQUESTS], the correspondence should always be addressed to the constituent.\n[/INST]\n\n[DOCUMENT]\n{document}\n[/DOCUMENT]\n\n[CONTEXT]\n{context}\n[/CONTEXT]\n\n[REQUESTS]\n{requests}\n[/REQUESTS]\n\n[PREVIOUS_RESPONSE]\n{previous_response}\n[/PREVIOUS_RESPONSE]",
    "type": "correspondence"
}`

Note modelId must match details entered in models table

File: app/routes/models.js [hunk 1]
```diff
@@ -1,77 +1,88 @@
...
+    if (models.length === 0) {
+      return h.response().code(204)
+    }
 
-      return h.response(models).code(200)
+    return h.response(models).code(200)
+  }
+},
+{
+  method: 'GET',
```
Comment (BarryGeeDefra) [inline app/routes/models.js:18]: is a separate route necessary here? the getModels() and getAllModels() functions look very similar. Would it not be more readable to have a single route with an optional deploymentVendor query param. When present only return those models and if not return all?

File: app/storage/repos/models.js [hunk 2]
```diff
@@ -86,9 +86,45 @@ const getModels = async (deploymentVendor) => {
...
+  const models = []
+
+  for await (const entity of query) {
+    models.push(entity)
+  }
+  models.sort((a, b) => a.PartitionKey - b.PartitionKey)
+
+  const allModels = []
+  let currentVendor = ''
+  let vendorModels = []
```
Comment (BarryGeeDefra) [inline app/storage/repos/models.js]: should be a const

**Review by BarryGeeDefra — COMMENTED** [review]
Just a couple of comments

File: app/storage/repos/models.js (hunk 2 above)
Comment (alistairknight) [reply app/storage/repos/models.js]: Good spot, sorted

File: app/routes/models.js (hunk 1 above)
Comment (alistairknight) [reply app/routes/models.js:18]: The JSON that is returned by both methods is different, `getModels()` is a simple array of models, whereas `getAllModels()` returns an array of vendors each with a nested array of models, so best to keep them separate

Comment (BarryGeeDefra) [reply app/routes/models.js:18]: That's fare. Any reason we need individual routes for that though? conditionally call getModels or getAllModels based on if the deploymentVendor param is provided? At the very least i would update getModels to getModelsByVendor and getAllModels to just getModels, or something along those lines. Just so its clear what the difference is at a glance.

Comment (alistairknight) [reply app/routes/models.js:18]: `getModels()` was the original behaviour, agree that `getModelsByVendor` might be more appropriate

Comment (alistairknight) [reply app/routes/models.js:18]: Commit [2952eca](https://github.com/DEFRA/coreai-mcu-prompts/pull/3/commits/2952ecaac03b365f346007a43ced1fe2d35737c6) renames routes

**Review by BarryGeeDefra — APPROVED** [review]
LGTM, there are build failures though. May just need a rerun.

**Pull request merged**